v5/v5_upgrade.ino

```diff
@@ -1119,13 +1108,90 @@ int findOccurrences(String str, String target) {
 }
 
 void wait(unsigned int seconds) { // most important task will be executed here
+  bool condition = false;
+  Println("Entering wait function...");
   for (int i = 0; (seconds > (i * 5)); i++) {
     if ((millis() / 1000) % terminationTime == 0 &&
         terminationTime > 0) // after every 5 minutes
     {
+      println("before termination");
       terminateLastMessage();
+      println("after termination");
+      condition = true;
+    }
+    if (displayWorking) {
+      if ((millis() / 1000) % 100 == 0) {
+        Println("before display status work in wait function");
+        messages_in_inbox =
+            totalUnreadMessages(); // ! will be depreciated in future versions
+        updateBatteryParameters(updateBatteryStatus());
+        condition = true;
+        Println("after display status work in wait function");
+      }
+    }
+    if (ThingSpeakEnable && wifiWorking &&
+        ((millis() / 1000) % updateInterval == 0)) {
+      if (wifi_connected()) {
+        Println("before thingspeak update");
+        updateThingSpeak(temperature, humidity);
+        Println("After thingspeak update");
+        condition = true;
+      }
+    }
+    if ((millis() / 1000) > 130 && DEBUGGING) {
+      Println("Disabling DEBUGGING...");
+      DEBUGGING = false;
+    }
+    if (((millis() / 1000) > 100) && (millis() / 1000) < 103) {
+      // it will just run at first time when system booted
+      Println("\nBefore updating values from message 1\n");
+      updateVariablesValues(readMessage(1));
+      Println("\nValues are updated from message 1\n");
+      delay(1000);
+      sendSMS("#setTime", "+923374888420");
       delay(1000);
+      i += 2000;
+      condition = true;
     }
-    delay(5);
+
+    if (condition) {
+      delay(1000);
+      condition = false;
+      i += 1000;
+    } else
+      delay(5);
   }
+  Println("leaving wait function...");
+}
+
+void setTime(String timeOfMessage) {
+  // +CMGL: 1,"REC READ","+923374888420","","23/08/14,17:21:05+20"
+  RTC = fetchDetails(timeOfMessage, "\"23/", "\"", 1);
+  println(
+      "\n+++++++++++++++++++++++++++++++++++++++++++\n Fetched data from : " +
+      timeOfMessage + "\nFinal data : " + RTC +
+      "\n+++++++++++++++++++++++++++++++++++++++++++\n");
+}
+
+void setTime() {
+  // this function will set Time struct using RTC string
```

Comment:
Time is updated in RTC struct once, so replace all delay functions and sync with this fucntion so the time will be updated in real time too